### DISCUSSION TITLE
Reading  custom fields.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#354] [Help] [answered]
I'm having the same issue as here: https://github.com/orgs/swellstores/discussions/152

But when I tried updating the permissions for my storefront's public key it's still not letting me read it.  I'm currently storing the value in the content namespace but I'd like to be able use it in the same way as `price` and add a few others as well.

I tried this changing the storefront public key permissions like so:

```
{
  "permissions": {
    "products": {
      "fields": ["price_per_person"]
      }
    }
}
```

It succeeded but didn't seem to update them, at least not in any way I could see. What am I doing wrong here?

![CleanShot 2023-08-24 at 12 40 54@2x](https://github.com/swellstores/community/assets/48241/b0b79951-bbd0-426a-90a1-6d3a9b48c359)

**Comment by logeshswell** [comment, marked as answer]
If you want to return the field (`price_per_person`) as normal as `price` field, then you need to create the field under **None** API namespace.
Then, granting the read permission like you described in the screenshot should work.

Can you try and confirm if that works?

**Reply by logeshswell** [reply]
Also, the recommended solution would be create the fields under `content` namespace only, as you don't need to set the permissions explicitly to access via the storefront APIs.

But in this case, you could access the value through `product.content.price_per_person` and not as `product.price_per_person`.

**Reply by holden** [reply]
Yes, that's what we've been doing. Throwing everything into the content namespace which works fine but I just to add this one as `product.price_per_person`. 

I changed it back to none (it was content previously), but it's still not accessible.

<img width="657" alt="CleanShot 2023-08-24 at 12 58 17@2x" src="https://github.com/swellstores/community/assets/48241/d385f9f7-e2ac-456a-9cb2-9da36024f923">

**Reply by holden** [reply]
Nevermind, it works fine, once I changed it back to none and give it a moment it started working.  Thanks!

**Reply by holden** [reply]
Apparently I wanted it both ways and to eat it too. ;-)

**Reply by holden** [reply]
I have a more complicated related question to permissions, what do we do about related collections?

We followed the tutorial here on adding vendors to products https://developers.swell.is/guides/build-a-marketplace. 

Now we just want to be able to expose a few fields on the product, eg the name of the vendor, their website, logo etc.

```
query getProductBySlug {
  productBySlug(slug: "evenchilada-feud") {
    id
    name
    slug
    price
      content {
      brand {
        name
      }
    }
  }
}
```

But I'm getting a permissions error "message": "This action is not permitted"

How does this work when the permissions are for accounts related to product?

**Reply by logeshswell** [reply]
If you open up permissions for the **accounts** model, then all the fields under **accounts** will be accessible. Otherwise, none of the field will be accessible.

I remember that we recently removed the support to open up permissions for individual fields for the related collections.

**Reply by holden** [reply]
Ok, I'll try that thanks. Is there anything I should or could do to limit the accounts exposed, eg scoping for an account group? If that were possible, or I suppose I could simply limited the fields I expose, name, description, etc.? instead of making all of the accessible.

**Reply by holden** [reply]
Last permission question I promise, I tried updating my account permissions on the public key to:

```
{
  "permissions": {
    "accounts": {
      "fields": [
        "name",
        "content"
      ]
    }
  }
}
```

Which seemed to work but I'm still getting permission errors, even though it nicely helps autocomplete as I type.

```
query getProductBySlug {
  productBySlug(slug: "evenchilada-feud") {
    id
    name
    slug
    price
    pricePerPerson
    content {
      brand {
        name
        content {
          websiteUrl
          description
          logo {
            data
          }
        }
      }
    }
  }
}
```

**Reply by holden** [reply]
@logeshswell I tried various things but I still cannot seem to get a connected model to work. It clearly see that it's connected correctly, it just doesn't like the permssions: "code": "permission_error",
<img width="343" alt="CleanShot 2023-08-25 at 11 07 29@2x" src="https://github.com/swellstores/community/assets/48241/7cffcefd-8b5e-438c-a3da-88a00d4bae7f">
